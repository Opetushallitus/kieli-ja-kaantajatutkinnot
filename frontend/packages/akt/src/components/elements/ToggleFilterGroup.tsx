import { CustomButton } from 'components/elements/CustomButton';
import { Color, Variant } from 'enums/app';

interface ToggleFilter<T> {
  status: T;
  label: string;
  count?: number;
  testId?: string;
}
interface ToggleFilterGroupProps<T> {
  filters: Array<ToggleFilter<T>>;
  activeStatus: T;
  onButtonClick(T: T): void;
}

export function ToggleFilterGroup<T>({
  filters,
  activeStatus,
  onButtonClick,
}: ToggleFilterGroupProps<T>) {
  return (
    <div className="columns">
      {filters.map(({ count, status, testId, label }, i) => (
        <CustomButton
          className="border-radius-unset"
          key={i}
          data-testid={testId}
          color={Color.Secondary}
          variant={
            activeStatus === status ? Variant.Contained : Variant.Outlined
          }
          onClick={() => onButtonClick(status)}
        >
          <div className="columns gapped">
            <div className="grow">{label}</div>
            {count && <div>{`(${count})`}</div>}
          </div>
        </CustomButton>
      ))}
    </div>
  );
}
