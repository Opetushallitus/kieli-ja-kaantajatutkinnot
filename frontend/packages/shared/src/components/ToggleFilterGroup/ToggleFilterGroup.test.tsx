import renderer from 'react-test-renderer';

import { ToggleFilterGroup } from './ToggleFilterGroup';

const enum FilterStatusEnum {
  Active,
  Passive,
  Expired,
}
const filterData = [
  {
    status: FilterStatusEnum.Active,
    label: 'Active',
    count: 10,
    testId: 'toggleFilter--active',
  },
  {
    status: FilterStatusEnum.Passive,
    label: 'Passive',
    count: 0,
  },
  {
    status: FilterStatusEnum.Expired,
    label: 'Expired',
  },
];

describe('ToggleFilterGroup', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <ToggleFilterGroup
          filters={filterData}
          activeStatus={FilterStatusEnum.Active}
          onButtonClick={jest.fn}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
