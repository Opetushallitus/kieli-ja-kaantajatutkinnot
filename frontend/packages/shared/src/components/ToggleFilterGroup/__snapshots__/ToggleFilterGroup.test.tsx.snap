// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleFilterGroup should render correctly 1`] = `
<div
  className="columns"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary border-radius-unset css-zcbmsk-MuiButtonBase-root-MuiButton-root"
    data-testid="toggleFilter--active"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="columns gapped"
    >
      <div
        className="grow"
      >
        Active
      </div>
      <div>
        (10)
      </div>
    </div>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary border-radius-unset css-1l9sosn-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="columns gapped"
    >
      <div
        className="grow"
      >
        Passive
      </div>
      <div>
        (0)
      </div>
    </div>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary border-radius-unset css-1l9sosn-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="columns gapped"
    >
      <div
        className="grow"
      >
        Expired
      </div>
    </div>
  </button>
</div>
`;
